hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/AnalyzeStatementTest.java

```diff
@@ -42,16 +44,54 @@ public static void beforeClass() {
     }
 
     @Test
-    public void test_select() {
+    public void test_options() {
         createMapping("test", Long.class, String.class);
         assertFalse(assertQueryPlan("SELECT * FROM test").isAnalyzed());
         assertTrue(assertQueryPlan("ANALYZE SELECT * FROM test").isAnalyzed());
+        // Default options
         SelectPlan plan = assertQueryPlan(
-                "ANALYZE WITH OPTIONS('opt1'='opt1val', 'opt2'='opt2val') SELECT * FROM test");
+                "ANALYZE WITH OPTIONS("
+                        + "'processingGuarantee'='exactlyOnce', "
+                        + "'snapshotIntervalMillis'='121', "
+                        + "'initialSnapshotName'='pressF', "
+                        + "'maxProcessorAccumulatedRecords'='100'"
+                        + ") SELECT * FROM test");
+        assertTrue(plan.isAnalyzed());
+
+        assertFalse(plan.analyzeJobConfig().isSplitBrainProtectionEnabled());
+        assertFalse(plan.analyzeJobConfig().isAutoScaling());
+        assertFalse(plan.analyzeJobConfig().isSuspendOnFailure());
+
+        assertTrue(plan.analyzeJobConfig().isMetricsEnabled());
+        assertTrue(plan.analyzeJobConfig().isStoreMetricsAfterJobCompletion());
+
+        assertEquals(ProcessingGuarantee.EXACTLY_ONCE, plan.analyzeJobConfig().getProcessingGuarantee());
+        assertEquals(121L, plan.analyzeJobConfig().getSnapshotIntervalMillis());
+        assertEquals("pressF", plan.analyzeJobConfig().getInitialSnapshotName());
+        assertEquals(100, plan.analyzeJobConfig().getMaxProcessorAccumulatedRecords());
+
+        // Test non-default options
+        plan = assertQueryPlan(
+                "ANALYZE WITH OPTIONS("
+                        + "'metricsEnabled'='false', "
+                        + "'storeMetricsAfterJobCompletion'='false'"
+                        + ") SELECT * FROM test");
         assertTrue(plan.isAnalyzed());
-        assertEquals("opt1val", plan.getAnalyzeOptions().get("opt1"));
-        assertEquals("opt2val", plan.getAnalyzeOptions().get("opt2"));
+        assertFalse(plan.analyzeJobConfig().isMetricsEnabled());
+        assertFalse(plan.analyzeJobConfig().isStoreMetricsAfterJobCompletion());
+    }
+
+    @Test
+    public void test_useUnsupportedOptionFails() {
+        createMapping("test", Long.class, String.class);
+        assertThatThrownBy(() -> sqlService.execute(
+                "ANALYZE WITH OPTIONS('splitBrainProtectionEnabled'='true') SELECT * FROM test"))
```

Comment:
we should test all options that should be unsupported, @burakgok was doing something similar for `ALTER JOB`